File: app/dbmodels/Cartridge.php
<?php

$table = array(
    'fields'        => array(),
    'oldfields'     => array(),
    'renamefields'  => array(),
    'indexes'       => array(),
    'oldindexes'    => array(),
    'relationships' => array(),
    'model'         => 'Cartridge',
    'menu'          => ''
);
$table['fields'] = array(
    'id'                 => DBModels::type('increments',
                                           array('visible' => false)),
    'entity_id'          => DBModels::type('integer'),
    'is_recursive'          => DBModels::type('boolean',
                                              array('visible' => false)),
    'cartridge_item_id'   => DBModels::type('integer'),
    'asset_id'           => DBModels::type('integer'),
    'date_in'            => DBModels::type('date'),
    'date_use'           => DBModels::type('date'),
    'date_out'           => DBModels::type('date'),
    'pages'              => DBModels::type('integer'),
);


$table['relationships'] = array(
    'cartridge_item' => array(
        'type' => 'belongsTo',
        'item' => 'CartridgeItem'),
    'asset' => array(
        'type' => 'belongsTo',
        'item' => 'Asset'),
    'entity' => array(
        'type'  => 'belongsTo',
        'item'  => 'Entity'),
);

File: fusionglpi/database/cartridge.py
from fusionglpi import db

class Cartridge(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    entity_id = db.Column(db.Integer, db.ForeignKey('entity.id'))
    is_recursive = db.Column(db.Boolean, nullable=False, server_default="0")
    cartridge_item_id = db.Column(db.Integer, db.ForeignKey('cartridge_item.id'))
    asset_id = db.Column(db.Integer, db.ForeignKey('asset.id'))
    date_in = db.Column(db.Date)
    date_use = db.Column(db.Date)
    date_out = db.Column(db.Date)
    pages = db.Column(db.Integer, nullable=False, server_default="0")
    
